ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -139,10 +140,10 @@ public class DiskManager {
     this.replicaStatusDelegates = replicaStatusDelegates;
     this.stoppedReplicas = stoppedReplicas;
     expectedDirs.add(reserveFileDir.getAbsolutePath());
+    diskMetrics = new DiskMetrics(storeMainMetrics.getRegistry(), disk.getMountPath(),
```

Comment:
not exactly, we don't have to create multiple disk metrics, that's why I only create one for it. But creating different disk metrics is not wrong, since each disk metric only has the references to the real histogram, counter metrics. There is only one metric being created in the MetricRegistry. And different disk metric objects are just referencing to the same metric.